### ISSUE TITLE
VSCode1.29.0 freezes when installing vscode-colorize

### ISSUE THREAD

**Issue by mokume** [KamiKillertO/vscode-colorize#174]
I updated to VSCode 1.29.0 today and VSCode freezes now.
I activated the extended function one by one, and I got on this extension function.

Since I am using CSS in JS, I enabled the following settings
```
"colorize.languages": [
  "javascript"
]
```
Also, I used Eslint

Perhaps when opening VSCode, colorize displays all javascript files. 
The reason is that ESLint warnings are displayed for all files which are not opened when VSCode is opened with colorize enabled.

I'm sorry, my English is not good

**Comment by KamiKillertO** [comment]
I'm sorry for the inconvenience 😞. Unfortunately there's nothing I can do anything for you yet ><. But I think you might be able to solve/reduce your problem by playing with the setting `colorize.exclude`. I suggest you to list your lib, tests, dependencies folders in this setting in order to remove them from the variable search that's trigger at boot time. 
I've also a question for you, did you use variables (for colors) in your code ? I've never have the occasion to play with CSS in JS so I'm not aware of the convention

**Comment by mokume** [comment]
Thank you for your reply.

I do not want to disable it with colorize.exclude.

I am sorry but I can not provide an execution environment right away.

I am using ReactNative for application development, and exactly I put the color code in the array of JS instead of CSS in JS.
There are thousands of JS files of my project.

The problem is that Colorize has been adapted to the file that has been opened so far.
When updating to VSCode1.29.0 today, when VSCode is opened, is not it scanning all the files of the format set in colorize.include in the project?
Therefore, a warning of ESLint of all the JS files in the project is displayed (maybe other extended functions may be used), it is thought that the CPU was used very much and it freezed.

As an image of the execution environment

- Enable ESLint
- Install 5 or 6 npm packages
- Target JS file of node_modules to Colorize
- Open VSCode
I think that I can reproduce with.

**Comment by KamiKillertO** [comment]
With the setting `colorize.exclude` you can "exclude" files. By excluding, I mean that the file will not be colorized and not opened by my extension. This setting is use only by the `colorize` extension not by vscode ^^ , adding files in this settings will not change anything elsewhere. I suggest you to add testing folder in this settings or files without styling in it.

**Comment by JonGarbayo** [comment]
Hi!

I have the same problem (VSCode 1.29.0 with colorize 0.8.0), and I use Ubuntu 18.04.
Due to the VSCode crash on boot, I had to downgrade to colorize 0.7.2 :/

My project uses SCSS and CSS, no CSS in JS, if it can help.

**Comment by KamiKillertO** [comment]
@JonGarbayo can you share you settings ? Only the settings starting by `colorize.`. Can you include default settings too ?

**Comment by mokume** [comment]
Hi!

I am using a Mac.
I made about 2,000 CSS files in the test.
In my environment I activated StyleLint.
We have opened VSCode with this setting. It did not completely freeze, but the motion got heavier.
Is it supposed to watch all target files again at startup?
"Code Helper" is kept around the utilization rate of 100 on the activity monitor.

@JonGarbayo 
How many SCSS and CSS do you have in your project?

[colorize-test.zip](https://github.com/KamiKillertO/vscode-colorize/files/2580235/colorize-test.zip)

**Comment by KamiKillertO** [comment]
Around 200 I think. 
I know what's happening but I don't know how I can solve it  😞. I've a quick fix in mind but it will "break" one major feature of the extension, the variables highlightings.

This is what happen when you open vscode:
- `colorize` start.
- `colorize` read the settings `colorize.languages`, `colorize.include` and `colorize.exclude`. With those settings `colorize` generate a list of globPattern that will be used to decide if a file should be colorized or not. 
- `colorize` get and read all files, matching the globPatterns, in order to find variables (css, sass...)
- `colorize` generate colorized background for colors for files visible in the editor.

Since vscode 1.29 open and read a file (from an extension) also trigger other extensions watching the file. So when `colorize` open the files in order to get variables in it, it also triggers linter's extensions like stylelint and eslint ><.

One quick fix would be to stop the variable search but it will prevent the display of css, sass... variables ><

**Comment by JonGarbayo** [comment]
@KamiKillertO What settings do you want: 0.7.2 or 0.8.0? And just to be sure, when you ask "default settings", you're also talking about default colorize settings?

@mokume I used a ``find`` command, and my project has 475 ``.css`` and 244 ``.scss`` files (excluding node_modules and all that stuff). I'm very surprised, I thought there was much more; it's an old and messy project x)

By the way, I tried with the newly released 1.29.1 version, but the problem is the same :(

**Comment by KamiKillertO** [comment]
@JonGarbayo I want you 0.8.0 settings ^^ and I mean colorize settings. I only need colorize settings.

I've also upgrade to vscode 1.29.1 and I'm facing the same issue 😿. I've got no way to fix this issue yet, I need to rethink the way i deal with variables (CSS, SASS...) or contact vscode team and find a solution with them.

**Comment by JonGarbayo** [comment]
OK, so here are my colorize 0.8.0 settings (yes, there are tons):
```json
"colorize.colorized_variables": [
    "CSS",
    "SASS",
    "LESS",
],
```

and the default ones:
```json
// You can specified the type of color that should be colorized by the extension.
"colorize.colorized_colors": [
  "BROWSERS_COLORS",
  "HEXA",
  "RGB",
  "HSL"
],

// You can specified the type of variables that should be colorized by the extension.
"colorize.colorized_variables": [
  "CSS"
],

// Configure glob patterns for excluding files and folders. Colorize will not colorized colors in these files and folders and it'll also not search for variables inside. Read more about glob patterns [here](https://code.visualstudio.com/docs/editor/codebasics#_advanced-search-options).
"colorize.exclude": [
  "**/.git",
  "**/.svn",
  "**/.hg",
  "**/CVS",
  "**/.DS_Store",
  "**/.git",
  "**/node_modules",
  "**/bower_components",
  "**/tmp",
  "**/dist",
  "**/tests"
],

// [deprecated] Specified a list of files extension that should be colorized. This setting is deprecated and will be removed soon, you should use the setting `colorize.include` instead.
"colorize.files_extensions": [],

// Hide colors for the current line. Set to `false` to turn it off
"colorize.hide_current_line_decorations": true,

// Ignores the warning for `colorize.files_extensions`.
"colorize.ignore_files_extensions_deprecation": false,

// Configure glob patterns for including files and folders. By default Colorize is enable for files matching one the languages defined in the `colorize.languages` config, with this config you can enable colorize for other files or folders. Read more about glob patterns [here](https://code.visualstudio.com/docs/editor/codebasics#_advanced-search-options).
"colorize.include": [],

// Configure a list of languages that should be colorized. You can learn about languages at https://code.visualstudio.com/docs/languages/overview.
"colorize.languages": [
  "css",
  "sass",
  "scss",
  "less",
  "postcss",
  "sss",
  "stylus",
  "xml",
  "svg"
]
```

**Comment by KamiKillertO** [comment]
Ok so by playing with the settings "colorize.languages", "colorize.include", "colorize.exclude" you might be able to reduce the problem. I see that you have setted colorization for `sass` and `less` is that normal ?

```
// Here list only the languages you use
"colorize.languages": [
  "css",
  "scss",
  "less"
],
// You can specified the type of variables that should be colorized by the extension.
"colorize.colorized_variables": [
  "CSS",
  "SASS",
  "LESS"
],

// Configure glob patterns for excluding files and folders. Colorize will not colorized colors in these files and folders and it'll also not search for variables inside. Read more about glob patterns [here](https://code.visualstudio.com/docs/editor/codebasics#_advanced-search-options).
"colorize.exclude": [
  "**/.git",
  "**/.svn",
  "**/.hg",
  "**/CVS",
  "**/.DS_Store",
  "**/.git",
  "**/node_modules",
  "**/bower_components",
  "**/tmp",
  "**/dist",
  "**/tests"
// you can add more here to reduce the number of files that might be opened
],
```

**Comment by suhao** [comment]
Dear,
I have the same problem.
I used VSCode(1.29.1, system setup) on windows10 + colorize 0.8.1.
The colorize 0.8.1 settings is the default ones.
The CPU always been 38%~40% and I always need to restart the VSCode.
Our project include the source code of [boost](https://www.boost.org/users/history/version_1_68_0.html), which had many html and xml files.

Maybe the colorize could be enabled just for the opened files?

**Comment by KamiKillertO** [comment]
I'm currently working on that, I'll have a settings to enable/disable the search for variables at opening time.

**Comment by KamiKillertO** [comment]
Hi,

I've release a new version with a new setting `enable_search_variables` that allow you to disable the variable search at opening time.

**Comment by JonGarbayo** [comment]
Hi,

I installed this new update, but the problem was the same. Though, it motivated me to investigate during my personal time to figure out what was wrong.
Now I know: the XML files of my project are really big (the biggest is like 300,000 lines long), and as the 0.8.x colorize versions are watching them by default, VS Code crashes.

So I solved the problem by removing the "xml" entry of the "colorize.languages" setting.

I hope it will help other people facing the same issue :)
